alexmenuet-alexmenuet-taulia/hw4/BookManager/app/controllers/stores_controller.rb

```diff
@@ -0,0 +1,61 @@
+class StoresController < ApplicationController
+    def index
+       @stores = Store.all
+       @new_store = Store.new
+       render "store_index"
+    end
+    
+    def edit
+        id = params[:id]
+        @store = Store.find(id)
+        render "store_edit"
+    end
+    
+    def create
+        @new_store = Store.new(store_params)
+        @new_store.save
+        redirect_to stores_path
+    end
+    
+    def update
+        store = params[:store]
+        @db_store = Store.find(store[:id])
+        @db_store.update(store_params)
+        redirect_to stores_path
+    end
+    
+    def show
+        @add_book = Book.new
+        @all_books = Book.new
+        store_id = params[:store_id]
+        @store = Store.find(store_id)
+        !searchbook
+        if @all_books
+            @all_books = Book.find_by_sql "SELECT b.* FROM books b WHERE b.id NOT in (SELECT j.book_id FROM joined_tables j WHERE j.store_id = #{store_id})"
+        else
```

Comment:
Have you found an alternative to writing raw SQL? ActiveRecord does a great job of abstracting SQL away from the developer's eyes.